app/controllers/users_controller.rb

```diff
@@ -1,7 +1,54 @@
 class UsersController < ApplicationController
   before_action :authenticate_user!
 
+  def index
+    if current_user.try(:admin?)
+      @users = User.all.order(created_at: :asc)
+    else
+      render(file: File.join(Rails.root, 'public/403.html'), status: 403, layout: false)
```

Comment:
iiiiIIIIIIiiiiinteresting. This is an unconventional way to do this, but is totally valid. Consider breaking some of this into private methods to make it less of a giant line.